tests/integration/api-gateway/gateway-tests/main_test.go

```diff
@@ -46,29 +48,46 @@ const jwtAndOauthOnePathApiruleFile = "jwt-oauth-one-path-strategy.yaml"
 const resourceSeparator = "---"
 const defaultHeaderName = "Authorization"
 
-var resourceManager *resource.Manager
+var (
+	resourceManager *resource.Manager
+	conf            Config
+	httpClient      *http.Client
+)
 
 type Config struct {
-	HydraAddr  string `envconfig:"TEST_HYDRA_ADDRESS"`
-	User       string `envconfig:"TEST_USER_EMAIL"`
-	Pwd        string `envconfig:"TEST_USER_PASSWORD"`
-	ReqTimeout uint   `envconfig:"TEST_REQUEST_TIMEOUT,default=180"`
-	ReqDelay   uint   `envconfig:"TEST_REQUEST_DELAY,default=5"`
-	Domain     string `envconfig:"DOMAIN"`
+	HydraAddr     string        `envconfig:"TEST_HYDRA_ADDRESS"`
+	User          string        `envconfig:"TEST_USER_EMAIL"`
+	Pwd           string        `envconfig:"TEST_USER_PASSWORD"`
+	ReqTimeout    uint          `envconfig:"TEST_REQUEST_TIMEOUT,default=180"`
+	ReqDelay      uint          `envconfig:"TEST_REQUEST_DELAY,default=5"`
+	Domain        string        `envconfig:"DOMAIN"`
+	ClientTimeout time.Duration `envconfig:"TEST_CLIENT_TIMEOUT,default=10s"` //Don't forget the unit!
+	IngressClient bool          `envconfig:"USE_INGRESS_CLIENT,default=false"`
```

Comment:
Maybe we should have an ENV that indicates local(minikube) environment instead of ingress client because later if needed we can perform more actions, not only configuring proper HTTP client. The `ingress client` is not straightforward, without looking into the implementation of the client you will not know why it is used (this is old implementation drawback)